horace_core/sqw/@sqw/calculate_qw_pixels2.m

```diff
@@ -79,13 +84,16 @@
     kf = ki;
 end
 
-qw = cell(1, 4);
-qw(1:3) = calculate_q (ki, kf, detdcn(:, idet), spec_to_rlu(:, :, irun));
+qw = cell(4, 1);
+qw(1:3) = calculate_q(ki, kf, detdcn(:, idet), spec_to_rlu(:, :, irun));
 qw{4} = eps_diff;
+% Join cell array into 4xN mat
+qw = cat(2, qw{:})';
+qw = win.data.proj.transform_hkl_to_pix(qw);
 
 end
 
-function [d_mat, detdcn] = spec_coords_to_det (detpar)
+function detdcn = spec_coords_to_det (detpar)
 % Matrix to convert coordinates in spectrometer (or laboratory) frame into detector frame
 %
 %   >> d_mat = spec_coords_to_det (detpar)
```

Comment:
This is code duplication as something similar is in `rundatah/convert_to_sqw`. Which one is better I do not know, as see some optimization here too. Probably other ticket again.

Reply:
A whole chunk of this including `calculate_qw_pixels2` is duplicated in `rundatah` (c.f. `rundatah.calc_qspec`) and the whole thing needs to be split off into utilities at some point, but like you say, not this ticket.